### PR TITLE
Adding library way

### PR DESCRIPTION
**Pull request by acaiafa**
@jtimberman @someara

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -1,48 +1,14 @@
...
-          :description  => 'IP address to use for memcached instance',
-          :default      => '0.0.0.0'
-
-attribute 'memcached/logfilename',
-          :display_name => 'Memcached logfilename',
-          :description  => 'The filename used to log memcached',
-          :default      => 'memcached.log'
+maintainer 'Anthony Caiafa'
+maintainer_email '2600.ac@gmail.com'
+license 'All rights reserved'
```
Comment (nathenharvey) [inline metadata.rb]: This is likely just an oversight but the license must remain Apache 2.0

File: metadata.rb [hunk 2]
```diff
@@ -1,48 +1,14 @@
...
-
-attribute 'memcached/listen',
-          :display_name => 'Memcached IP Address',
-          :description  => 'IP address to use for memcached instance',
-          :default      => '0.0.0.0'
-
-attribute 'memcached/logfilename',
-          :display_name => 'Memcached logfilename',
-          :description  => 'The filename used to log memcached',
-          :default      => 'memcached.log'
```
Comment (nathenharvey) [inline metadata.rb]: Why not have supported attributes listed?

Comment (someara) [inline metadata.rb]: Attributes in the metadata are a bit of an anacronism..

They're not consumed by anything,they require tedious duplication between
the metadata.rb file and the attributes files, and become inaccurate if
there is any kind of load-time decision logic happening

-s

**Comment by acaiafa** [comment]
@tas50 want to take a look at this?

**Comment by acaiafa** [comment]
too much clicking

**Comment by stonith** [comment]
@acaiafa Haven't had time to really test this but it's missing a bunch of memcached options that were added in 1.8.0.

**Comment by acaiafa** [comment]
@stonith pretty simple fix. I can add them directly OR you can you show me a README for these additional memcached options? 

I can always add in a simple options_collector and that will solve this also.

**Comment by stonith** [comment]
@acaiafa https://github.com/chef-cookbooks/memcached lists the additional options.

**Comment by acaiafa** [comment]
Ok i mean thats a small subset of options. I will push this up later. Can you just examine the bulk of the work and see if it fits.

**Comment by tas50** [comment]
Anthony,

I think the spirit of this PR has been implemented in the new release.  Users still have the opportunity to install memcached under the system system's init system, but the old definition has been replaced with a 12.5 custom resource that behaves as it should.  The mix of attributes and definition is gone, which makes it more of a true library cookbook.  There is a current restart bug that I'm working to track down, but let me know if this solves the need for a library cookbook.

-Tim

**Pull request closed without merging**